### PR TITLE
add election backtest

### PR DESCRIPTION
**Pull request by zgfzgf**
./lotus-shed election backtest t01000
height, winCount
223, 5
222, 10
221, 5
220, 7
219, 5
... ...
./lotus-shed election backtest -h
NAME:
   lotus-shed election backtest - Backtest elections with given miner

USAGE:
   lotus-shed election backtest [command options] [minerAddress]

OPTIONS:
   --height value  blockchain head height (default: 0)
   --count value   blockchain count (default: 120)
   --help, -h      show help (default: false)

### CODE REVIEW

File: cmd/lotus-shed/election.go [hunk 1]
```diff
@@ -124,3 +130,100 @@ var electionEstimate = &cli.Command{
...
+
+	winner, err := gen.IsRoundWinner(ctx, ts, round, miner, brand, mbi, api)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to check if we win next round: %w", err)
+	}
+
+	if winner == nil {
+		return nil, nil
+	}
+	return winner, nil
```
Comment (magik6k) [inline cmd/lotus-shed/election.go:228]: Could just be
```suggestion
	return winner, nil
```

File: cmd/lotus-shed/election.go [hunk 2]
```diff
@@ -124,3 +130,100 @@ var electionEstimate = &cli.Command{
...
+	Usage:     "Backtest elections with given miner",
+	ArgsUsage: "[minerAddress]",
+	Flags: []cli.Flag{
+		&cli.Uint64Flag{
+			Name:  "height",
+			Usage: "blockchain head height",
+		},
+		&cli.IntFlag{
+			Name:  "count",
+			Usage: "blockchain count",
```
Comment (magik6k) [inline cmd/lotus-shed/election.go:145]:
```suggestion
			Usage: "number of won elections to look for",
```

**Review by magik6k — COMMENTED** [review]
2 nits, but this looks useful.

It would also be great to check if the miner had a block in tipsets in which it was eligible to mine, and report if it wasn't present.

File: cmd/lotus-shed/election.go (hunk 2 above)
Comment (zgfzgf) [reply cmd/lotus-shed/election.go]: now OK

File: cmd/lotus-shed/election.go (hunk 1 above)
Comment (zgfzgf) [reply cmd/lotus-shed/election.go]: thank you.

**Review by arajasek — APPROVED** [review]

**Comment by arajasek** [comment]
Thanks for this!

**Pull request merged**